### PR TITLE
Added Option for AP Build URL and sanity marker moved to 'sanity and …

### PR DESCRIPTION
**Pull request by shivamcandela**
…not interop'

Signed-off-by: shivamcandela <shivam.thakur@candelatech.com>

### CODE REVIEW

File: .github/workflows/sanity.yml [hunk 1]
```diff
@@ -27,6 +27,10 @@ on:
         default: 'sanity'
         description: 'Pytest marker expression that will be used to select the tests to execute'
         required: false
+      build_url:
+        default: '0'
```
Comment (brennerm) [inline .github/workflows/sanity.yml:31]: What does a default value of 0 mean? That's pretty hard to understand. I'd prefer to default to an empty string and explain in the description what happens in this case.

**Comment by brennerm** [comment]
@shivamcandela Please also adjust these two conditions:
https://github.com/Telecominfraproject/wlan-testing/blob/c0ffb8b954fc8e0f6433a7d62e4e10b11c939825/.github/workflows/sanity.yml#L310
https://github.com/Telecominfraproject/wlan-testing/blob/c0ffb8b954fc8e0f6433a7d62e4e10b11c939825/.github/workflows/sanity.yml#L316

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (shivamcandela) [reply .github/workflows/sanity.yml:31]: '0' is the default string is given in ini file as well but we can give any value

File: .github/workflows/sanity.yml [hunk 2]
```diff
@@ -183,7 +187,7 @@ jobs:
                   - -c
                   - |
                     cd tests
-                    pytest -m "${{ github.event.inputs.marker_expression || 'sanity' }}" -s -vvv --testbed="${{ matrix.testbed }}" --skip-testrail --alluredir=/tmp/allure-results
+                    pytest -m "${{ github.event.inputs.marker_expression || 'sanity and not interop' }}" -s -vvv --testbed="${{ matrix.testbed }}" --skip-testrail --alluredir=/tmp/allure-results -o --build=${{ github.event.inputs.build_url }}
```
[the remark this answers is not in the document]
Comment (shivamcandela) [reply .github/workflows/sanity.yml:190]: what is this fallback expression?

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (brennerm) [reply .github/workflows/sanity.yml:31]: Again, that won't help people using this workflow. Please change this value to something more meaningful and/or add some explanation to the description.

File: .github/workflows/sanity.yml [hunk 3]
```diff
@@ -183,7 +187,7 @@ jobs:
                   - -c
                   - |
                     cd tests
-                    pytest -m "${{ github.event.inputs.marker_expression || 'sanity' }}" -s -vvv --testbed="${{ matrix.testbed }}" --skip-testrail --alluredir=/tmp/allure-results
+                    pytest -m "${{ github.event.inputs.marker_expression || 'sanity and not interop' }}" -s -vvv --testbed="${{ matrix.testbed }}" --skip-testrail --alluredir=/tmp/allure-results -o --build=${{ github.event.inputs.build_url || "0"}}
```
Comment (brennerm) [inline .github/workflows/sanity.yml:190]: Think the parameter name should be a bit more expressive. `build` can mean a lot of things. Would prefer something like `ap_firmware_url`

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (shivamcandela) [reply .github/workflows/sanity.yml]: done

File: .github/workflows/sanity.yml (hunk 3 above)
Comment (shivamcandela) [reply .github/workflows/sanity.yml]: done

File: .github/workflows/sanity.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (shivamcandela) [reply .github/workflows/sanity.yml]: done

**Review by brennerm — APPROVED** [review]

**Pull request merged**